site/docs/reference/Connectors/materialization-connectors/Rockset.md

```diff
@@ -42,4 +42,123 @@ materializations:
       	workspace: ${namespace_name}
       	collection: ${table_name}
     source: ${tenant}/${source_collection}
+```
+## Bulk ingestion for large backfills of historical data
+
+You can backfill large amounts of historical data into Rockset using a *bulk ingestion*. Bulk ingestion must originate in S3 and requires additional steps in your dataflow. Flow's Rockset connector supports this through the GitOps workflow.
+
+### Prerequisites
+Before completing this workflow, make sure you have:
+* A working catalog spec including at least one Flow collection with its schema specified.
+* A production or [development](../../../getting-started/installation.md) environment
+::: Tip
+The following is an intermediate workflow. As needed, refer to [this guide](../../../guides/create-dataflow.md) for the basic steps to create and deploy a catalog spec using the GitOps workflow.
+:::
+
+### How to configure a bulk ingestion
+If you already have an [S3 integration](https://rockset.com/docs/amazon-s3/) in Rockset, you can simply direct the connector to it. The connector will create the collection from the given integration, process all objects in the S3 bucket first, and then begin writing additional any additional, newer data that is coming through your pipeline. This ensures that documents are always written to Rockset in the proper order.
```

Comment:
I think this is sorta right in terms of the conceptual understanding of how the connector itself works, but it's not a use case we should document here. The garden path is to start with all your data in a Flow collection, and wanting to materialize that into Rockset, which requires the two step process. _Technically_ you could imagine some use case where you have some other process (besides Flow) that's populating an S3 bucket that you want to bulk ingest, but even in that case, it's not clear how that data should be ordered with respect to the data in the Flow collection. Basically, it'd be a pretty "advanced" use case, with a number of foot-guns.

I think a better approach in terms of this doc would be to explain (maybe at the bottom, even) that you can always create the Rockset collection's yourself, using whatever parameters you want, and that the connector will simply use that collection as long as it has the same name. Basically, documenting that there's an "escape hatch" for Rockset power users. In that case, one _could_ create the collection using whatever integration, and decide to also name the integration in the Flow connector config _if_ they need it await completion of the integration before materializing.

Reply:
Ohhhkay. Thank you for reframing all of this for me. I see what you're saying. Given that, it's probably best to not document it at all. It's still possible, and potentially an advanced Rockset user would figure it out and try it on their own. There's always workarounds like that... but as the company we should only document the recommended workflow.